crates/kubelet/src/handle.rs

```diff
@@ -33,27 +34,123 @@ pub trait Stop {
     async fn wait(&mut self) -> anyhow::Result<()>;
 }
 
+/// Trait to describe necessary behavior for creating multiple log readers.
+/// TODO: Both providers make a handle containing a tempfile. If this is a common pattern,
+/// it might make sense to provide that implementation here. This would add `tempfile` as a
+/// dependency of `kubelet`.
+pub trait LogHandle<R>: Sync + Send {
```

Comment:
So after looking at the code, do we need to use this trait? We already have an `R` that is constrained with the proper traits, so couldn't we just reuse that, avoid the `Box` altogether and then just have `stream_logs` accept `Lines<BufReader>`? No matter what the implementation, it should always be a reader of some type, so we should be able to get a BufReader. 

@rylev any thoughts here?

Reply:
I ran into a couple of issues here.

1. This does not allow you to service two log requests at once because you would have to borrow / transfer ownership of the reader to the scope servicing the first request. 
2. Enforcing that R is clone is actually not very easy, for instance tokio::fs::File is not clone. 
3. Even if you can have two tasks reading from the same async reader, a new connection would cause it to seek and other clients would receive different sections of the reader than they were expecting. 

So at the very least we need to know how to create new AsyncReaders on demand, and they need to be Sync + Send to get spawned into the streaming function.

Reply:
I don't think issue 1 and 2 are necessarily impossible to overcome, but issue 3 is definitely an issue. I think the solution we have here isn't a bad one although our RuntimeHandle is becoming _very_ generic which makes it hard to reason about.

Reply:
Ok, so let's keep this as is. We can revisit it later as needed